### PR TITLE
Core: Structure deprecations with additional metadata

### PR DESCRIPTION
**Pull request by thommay**
Primarily adding URLs so that deprecation messages can point to full
documentation on the deprecation and how to fix it.

cc @lamont-granquist @tas50

### CODE REVIEW

**Comment by thommay** [comment]
@coderanger this.

**Comment by thommay** [comment]
Goes rather hand in hand with chef/chef-web-docs#231

**Comment by thommay** [comment]
@chef/client-core @adamleff

File: lib/chef/deprecated.rb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    class InternalApi < Base
+      def id
+        0
+      end
+      def target
+        "internal_api.html"
+      end
+    end
+
+3694
```
Comment (lamont-granquist) [inline lib/chef/deprecated.rb]: derp?

Comment (thommay) [inline lib/chef/deprecated.rb]: derp indeed.

File: lib/chef/deprecated.rb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    class InternalApi < Base
+      def id
+        0
+      end
+      def target
+        "internal_api.html"
+      end
+    end
+
```
Comment (thommay) [inline lib/chef/deprecated.rb]: Yeah, de-duped now :)

**Comment by adamleff** [comment]
Not a maintainer, but 👍 from me! The changes to the data collector stuff will work just fine for us too. Well done, @thommay!

**Pull request merged**